tests/contrib/operators/test_gcp_bigtable_operator.py

```diff
@@ -0,0 +1,526 @@
+    @mock.patch('airflow.contrib.operators.gcp_bigtable_operator.BigTableHook')
+    def test_empty_attribute(self, missing_attribute, project_id, instance_id, table_id, mock_hook):
+        with self.assertRaises(AttributeError) as e:
+            BigTableTableWaitForReplicationSensor(
+                project_id=project_id,
+                instance_id=instance_id,
+                table_id=table_id,
+                task_id="id"
+            )
+        err = e.exception
+        self.assertEqual(str(err), 'Empty parameter: {}'.format(missing_attribute))
+        mock_hook.assert_not_called()
+
+    @mock.patch('airflow.contrib.operators.gcp_bigtable_operator.BigTableHook')
+    def test_wait_no_instance(self, mock_hook):
+        mock_hook.return_value.get_instance.return_value = None
+
+        op = BigTableTableWaitForReplicationSensor(
+            project_id=PROJECT_ID,
+            instance_id=INSTANCE_ID,
+            table_id=TABLE_ID,
+            task_id="id"
+        )
+        self.assertFalse(op.poke(None))
+        mock_hook.assert_called_once_with()
+
+    @mock.patch('airflow.contrib.operators.gcp_bigtable_operator.BigTableHook')
+    def test_wait_no_table(self, mock_hook):
+        mock_hook.return_value.get_instance.return_value = mock.Mock(Instance)
+        mock_hook.return_value.get_cluster_states_for_table.side_effect = mock.Mock(
+            side_effect=google.api_core.exceptions.NotFound("Table not found."))
+
+        op = BigTableTableWaitForReplicationSensor(
+            project_id=PROJECT_ID,
+            instance_id=INSTANCE_ID,
+            table_id=TABLE_ID,
+            task_id="id"
+        )
+        self.assertFalse(op.poke(None))
+        mock_hook.assert_called_once_with()
+
+    @mock.patch('airflow.contrib.operators.gcp_bigtable_operator.BigTableHook')
+    def test_wait_not_ready(self, mock_hook):
+        mock_hook.return_value.get_instance.return_value = mock.Mock(Instance)
+        mock_hook.return_value.get_cluster_states_for_table.return_value = {
+            "cl-id": ClusterState(0)
+        }
+        op = BigTableTableWaitForReplicationSensor(
+            project_id=PROJECT_ID,
+            instance_id=INSTANCE_ID,
+            table_id=TABLE_ID,
+            task_id="id"
+        )
+        self.assertFalse(op.poke(None))
+        mock_hook.assert_called_once_with()
+
+    @mock.patch('airflow.contrib.operators.gcp_bigtable_operator.BigTableHook')
+    def test_wait_ready(self, mock_hook):
+        mock_hook.return_value.get_instance.return_value = mock.Mock(Instance)
+        mock_hook.return_value.get_cluster_states_for_table.return_value = {
+            "cl-id": ClusterState(4)
+        }
+        op = BigTableTableWaitForReplicationSensor(
+            project_id=PROJECT_ID,
+            instance_id=INSTANCE_ID,
+            table_id=TABLE_ID,
+            task_id="id"
+        )
+        self.assertTrue(op.poke(None))
+        mock_hook.assert_called_once_with()
```

Comment:
Add integration tests for the whole DAG.

Reply:
This is more complexed than I initially thought. Problem is that example DAG runs tasks in parallel and tests are ran on `SequentialExecutor` which doesn't allow concurrency. And it's not straightforward to alter this. 

I'll be happy to add it later.